domain/blockdevice/service/service.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/core/watcher/eventsource"
+	"github.com/juju/juju/domain/blockdevice"
+	"github.com/juju/juju/domain/blockdevice/state"
+)
+
+// State defines an interface for interacting with the underlying state.
+type State interface {
+	BlockDevices(ctx context.Context, machine string) ([]blockdevice.BlockDevice, error)
```

Comment:
This has come up a lot in #16601 around what id to take down into the state layer. The consensus at the moment is that the UUID should be what is going to the state layer. You have access to it when using this function so I would suggest we swap over to it.